### PR TITLE
Add warning feature to linux gcc toolchain

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Add warning feature to linux gcc toolchain

### Testing Done
CI

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by jonathanlew** [comment]
Will re-merge after #1256 gets merged

**Comment by jonathanlew** [comment]
Ready for review!

File: src/cc_toolchain/cc_toolchain_config.bzl [hunk 1]
```diff
@@ -540,6 +540,7 @@ def _linux_gcc_impl(ctx):
             "builtin_include_directories",
             "c++17",
             "determinism",
+            "warnings",
```
Comment (akhilveeraghanta) [inline src/cc_toolchain/cc_toolchain_config.bzl:543]: :warning:

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
I don't like this word

**Review by akhilveeraghanta — APPROVED** [review]
jk LG:tm: :shipit:

**Review by MathewMacDougall — APPROVED** [review]
`#warning LGTM`

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Pull request merged**